### PR TITLE
file-chooser: Add "directory" option

### PR DESCRIPTION
**Pull request by hadess**
Add a new directory option to "OpenFile" to allow selecting directories
instead of files.

### CODE REVIEW

**Comment by hadess** [comment]
WIP for https://github.com/flatpak/xdg-desktop-portal/issues/455

File: src/file-chooser.c [hunk 1]
```diff
@@ -439,6 +439,7 @@ static XdpOptionKey open_file_options[] = {
   { "accept_label", G_VARIANT_TYPE_STRING, NULL },
   { "modal", G_VARIANT_TYPE_BOOLEAN, NULL },
   { "multiple", G_VARIANT_TYPE_BOOLEAN, NULL },
+  { "directory", G_VARIANT_TYPE_BOOLEAN, NULL },
```
Comment (matthiasclasen) [inline src/file-chooser.c]: This means we'll pass the option to the backend, so we should probably update the backend interface description at the same time to also mention that option.

**Comment by matthiasclasen** [comment]
One thing I wondered is: would it be nicer to make this a separate OpenDirectory method ?

**Comment by hadess** [comment]
> One thing I wondered is: would it be nicer to make this a separate OpenDirectory method ?

I wondered that as well, but apart from the file filters, you'd probably want pretty much all the same options there. Really up to you, at the end of the day, it'd be hidden behind the GtkFileChooserNative API for app developers. Maybe @grulja strongly prefers a separate method?

**Comment by grulja** [comment]
> > One thing I wondered is: would it be nicer to make this a separate OpenDirectory method ?
> 
> I wondered that as well, but apart from the file filters, you'd probably want pretty much all the same options there. Really up to you, at the end of the day, it'd be hidden behind the GtkFileChooserNative API for app developers. Maybe @grulja strongly prefers a separate method?

In QFileDialog it's basically same, it's same method, just a different flag so I don't have any preference. The QFileDialog has following flags:
`AnyFile, ExistingFile, Directory, ExistingFiles` 
Which means that even for the "multiple" option, it's just a matter of flag combination.

**Comment by hadess** [comment]
Hopefully it works, because I still haven't had time to test it...

**Pull request merged**